vllm/entrypoints/openai/api_server.py

```diff
@@ -1877,6 +1879,14 @@ async def run_server_worker(listen_address,
         vllm_config = await engine_client.get_vllm_config()
         await init_app_state(engine_client, vllm_config, app.state, args)
 
+        nixl_side_channel_server = None
+        try:
+            nixl_side_channel_server = await \
+                set_up_nixl_side_channel_server(vllm_config)
+        except Exception as e:
+            logger.error("Failed to start NIXL side channel server: %s", e)
+            raise
```

Comment:
I'm unsure about having nixl-specific stuff here. We should probably think of it more as a generic way for connectors to expose their metadata on an http endpoint.

Also we need to be careful that the copy of the config here is the same one that gets update with the metadata during the init handshake. I'm not sure that will be the case when there are multiple API servers for example, we may need some additional coordination logic to ensure the info is passed in to each of the API servers.